src/mod.test.ts

```diff
@@ -538,6 +538,48 @@ if (typeof WebAssembly === 'undefined') {
         await plugin.close();
       }
     });
+
+    // TODO(chrisdickinson): this turns out to be pretty tricky to test, since
+    // deno and node's wasi bindings bypass JS entirely and write directly to
+    // their respective FDs. I'm settling for tests that exercise both behaviors.
+    test('when EXTISM_ENABLE_WASI_OUTPUT is not set, WASI output is stifled', async () => {
+      if ((globalThis as unknown as any).process) {
+        (
+          globalThis as unknown as Record<string, { env: Record<string, string> }>
+        ).process.env.EXTISM_ENABLE_WASI_OUTPUT = '';
+      } else if ((globalThis as unknown as any).Deno) {
+        globalThis.Deno.env.set('EXTISM_ENABLE_WASI_OUTPUT', '');
+      }
+      const plugin = await createPlugin('http://localhost:8124/wasm/wasistdout.wasm', {
+        useWasi: true,
+      });
+
+      try {
+        await plugin.call('say_hello');
+      } finally {
+        await plugin.close();
+      }
+    });
+
+    test('respects enableWasiOutput', async () => {
+      if ((globalThis as unknown as any).process) {
+        (
+          globalThis as unknown as Record<string, { env: Record<string, string> }>
+        ).process.env.EXTISM_ENABLE_WASI_OUTPUT = '';
+      } else if ((globalThis as unknown as any).Deno) {
+        globalThis.Deno.env.set('EXTISM_ENABLE_WASI_OUTPUT', '');
+      }
+      const plugin = await createPlugin('http://localhost:8124/wasm/wasistdout.wasm', {
+        useWasi: true,
+        enableWasiOutput: true,
```

Comment:
just to confirm, to get WASI output the host user needs to set `EXTISM_ENABLE_WASI_OUTPUT` and include `enableWasiOutput: true` here?

Reply:
Ah, not quite! The option is more specific than the env var. The env var is consulted when the option isn't specifically provided during plugin creation. If the env var is set to true, a user can still explicitly squelch output by passing `enableWasiOutput: false` to a specific `createPlugin` call, & vice versa.

(I added the option primarily for the browser target: env vars are usually resolved there furing bundle time, but may require bundler plugins to do so. I wanted to ensure we're as bundler-agnostic as possible.)

Reply:
oh right, great call to support the browser env! thank you